# HG changeset patch
# User twisti
# Date 1302771192 25200
# Node ID 1fcd6e9c3965f4bf23f6a8f4fbb77fdf74de37a7
# Parent  3a808be061ffc10da5647e50cc28252d70f21883
7036220: Shark fails to find LLVM 2.9 System headers during build
Reviewed-by: gbenson, twisti
Contributed-by: Xerxes Ranby <xerxes@zafena.se>

diff -r 3a808be061ff -r 1fcd6e9c3965 src/share/vm/shark/llvmHeaders.hpp
--- openjdk/hotspot/src/share/vm/shark/llvmHeaders.hpp	Wed Apr 13 14:33:03 2011 -0700
+++ openjdk/hotspot/src/share/vm/shark/llvmHeaders.hpp	Thu Apr 14 01:53:12 2011 -0700
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 1999, 2010, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 1999, 2011, Oracle and/or its affiliates. All rights reserved.
  * Copyright 2008, 2009, 2010 Red Hat, Inc.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
@@ -46,7 +46,11 @@
 #include <llvm/ModuleProvider.h>
 #endif
 #include <llvm/Support/IRBuilder.h>
+#if SHARK_LLVM_VERSION >= 29
+#include <llvm/Support/Threading.h>
+#else
 #include <llvm/System/Threading.h>
+#endif
 #include <llvm/Target/TargetSelect.h>
 #include <llvm/Type.h>
 #include <llvm/ExecutionEngine/JITMemoryManager.h>
@@ -55,8 +59,12 @@
 #include <llvm/ExecutionEngine/JIT.h>
 #include <llvm/ADT/StringMap.h>
 #include <llvm/Support/Debug.h>
+#if SHARK_LLVM_VERSION >= 29
+#include <llvm/Support/Host.h>
+#else
 #include <llvm/System/Host.h>
 #endif
+#endif
 
 #include <map>
 
